resource/opentelemetry-resource-detector-azure/test-requirements.txt

```diff
@@ -8,5 +8,8 @@ pytest==7.4.4
 typing_extensions==4.12.2
 wrapt==1.16.0
 zipp==3.19.2
--e opentelemetry-instrumentation
+opentelemetry-api==1.21  # when updating, also update in pyproject.toml
```

Comment:
Good catch on this. I think min version would be [v1.23](https://github.com/open-telemetry/opentelemetry-python-contrib/releases/tag/v0.44b0) for v0.44b0 for `opentelemetry-instrumentation`.